### ISSUE TITLE
oc problems after upgrading to Mac Sierra 

### ISSUE THREAD

**Issue by rawlingsj** [openshift/origin#11085]
The latest Mac OS update 'Sierra' is causing [problems with golang binaries compiled with less than go 1.7.1](https://github.com/golang/go/issues/16570).  gofabric8 was hit by this but after compiling with 1.7.1 and releasing the problems seems to be resolved. 

Problems running oc commands: deleting deployment configs sometimes hangs, `oc get pods` and `oc edit dc foo` intermittently causing panic errors, see the errors in gist...

- oc get pod https://gist.github.com/rawlingsj/8f732478abc5f9055e856cf9325224bc
- oc edit https://gist.github.com/rawlingsj/e2b5db98a3e2e31a8cb6262dc80a7d62

##### Version
```
oc v1.3.0
kubernetes v1.3.0+52492b4
features: Basic-Auth
kubernetes v1.2.0-36-g4a3f9c5
```

##### Steps To Reproduce
1. upgrade Mac to Sierra http://www.apple.com/uk/macos/sierra/
2. connect to openshift cluster, I was using an openshift dedicated instance
3. start interacting with the cluster using CLI commands as described above

**Comment by liggitt** [comment]
Looks like go 1.6.3 should fix this
https://github.com/golang/go/issues/16354

**Comment by rawlingsj** [comment]
Unfortunately I don't think that's going to be enough, our gofabric8 client was compiled with 1.6.3 when we had the issue reported.  1.7.1 has fixed it though.

**Comment by liggitt** [comment]
That's unfortunate and should probably be reported upstream if they think 1.6.3 resolved Sierra issues

**Comment by jimmidyson** [comment]
@rawlingsj We should probably check with all components compiled with 1.6.3 before reporting - it looks like it should work.

**Comment by jimmidyson** [comment]
Think they know it wasn't fixed in 1.6.3 - https://github.com/golang/go/issues/16354#issuecomment-236995514

**Comment by fabianofranz** [comment]
@eparis @smarterclayton arguments for 1.7.1?

**Comment by fabianofranz** [comment]
This will likely bit everyone in Sierra on Origin, Online, and OSE 1.3, 3.3. I'm increasing the priority and we have to debate about a new 0.x release or just 3.3.1. @pweil- @smarterclayton @juanvallejo

**Comment by fabianofranz** [comment]
cc @jwforres

**Comment by liggitt** [comment]
opened https://github.com/golang/go/issues/17234 to track backporting the final version of the fix for sierra to 1.6.x, we'll see if it happens

**Comment by fabianofranz** [comment]
Upstream issue is https://github.com/kubernetes/kubernetes/issues/32999.

**Comment by rafabene** [comment]
This bug is so unfortunate! I just updated MacOS to Sierra and now I can't use OpenShift/Kubectl anymore. @fabianofranz Do you have any updates on this bug? Any "urgent" release is planned?

**Comment by rafabene** [comment]
**WORKAROUND** for people who already updated to Sierra: 
Use the brew version: `brew install openshift-cli` It is already compiled for Sierra.

**Comment by fabianofranz** [comment]
@rafabene we're targeting 3.3.1 / 1.3.1. This is related to an [issue](https://github.com/golang/go/issues/16570) with Go which can be reproduced on mostly anything compiled by Go < 1.7 and used on MacOS Sierra (including [fabric8](https://github.com/fabric8io/fabric8/issues/6382), [kubectl](https://github.com/kubernetes/kubernetes/issues/32999) for example). Our strategy has been to work on a port of the fix to 1.6.x on Fedora/RHEL which we will then use to compile the .1 release. 

Meanwhile, the workaround is to compile `oc` and `kubectl` by yourself using Go 1.7.x (or use a tool that does that for you, like `brew` as you mentioned).

**Comment by fabianofranz** [comment]
Fixed in 1.3.1 and 3.3.1.
